tests/rptest/services/redpanda.py

```diff
@@ -1481,37 +1538,41 @@ class RedpandaServiceCloud(RedpandaServiceK8s):
     GLOBAL_CLOUD_CLUSTER_CONFIG = 'cloud_cluster'
 
     def __init__(self,
-                 context,
-                 num_brokers,
+                 context: TestContext,
                  *,
+                 config_profile_name: str,
                  superuser: Optional[SaslCredentials] = None,
                  skip_if_no_redpanda_log: Optional[bool] = False,
-                 tier_name: Optional[str] = None,
+                 min_brokers: int = 3,
                  **kwargs):
         """Initialize a RedpandaServiceCloud object.
 
         :param context: test context object
         :param num_brokers: ignored because Redpanda Cloud will launch the number of brokers necessary to satisfy the product needs
         :param superuser:  if None, then create SUPERUSER_CREDENTIALS with full acls
         :param tier_name:  the redpanda cloud tier name to create
+        :param min_brokers: the minimum number of brokers the consumer of the cluster (a test, typically) requires
+                            if the cloud cluster does not have this many brokers an exception is thrown
         """
 
-        super(RedpandaServiceCloud,
-              self).__init__(context,
-                             None,
-                             cluster_spec=ClusterSpec.empty(),
-                             superuser=superuser,
-                             skip_if_no_redpanda_log=skip_if_no_redpanda_log)
-        if num_brokers is not None:
-            self.logger.info(
-                f'num_brokers is {num_brokers}, but setting to None for cloud')
+        # we save the test context under both names since RedpandaService and Service
+        # save them under these two names, respetively
+        self.context = self._context = context
```

Comment:
It's possible, but is there much of a concrete benefit? This ABC appears a hierarchy which always provides a `.context` member. This is also how the KubeServiceMixin behaves, and similarly any other mixin I add.

If we pass it, we will need to store it, I guess in `self.__context` to make it class-private to avoid clobbering the context var from other classes. The behavior of the constructors for the Service hierarchy is not totally straightforward since on the other flavor (`RedpandaService`) we inherit from `Service` which isn't really a cooperating base class and wants `context` as the first positional argument. So run into the mixin/multiple inheritance issue where you need to avoid passing `context` to a base that doesn't expect it. `object` doesn't expect it, for example. So you need to then pass `context` _twice_ from `RedpandaService`: once as the first positional argument to be consumed (eventually) by `Service` and once under some other name to be consumed by `RedpandaServiceABC`. 

All for little gain I think. The mixin pattern mostly hides the types from the type-checker anyway (as we are often forwarding **kwargs, essentially erasing the types), which I think is the main benefit of passing it.

Other options are possible, e.g., the ABC could declare an abstract method or property returning a context which the bases would have to implement. This bypasses the constructor.

Despite this wall of text, I'm not convinced one way or the other, so interested to hear your thoughts too.